crates/metal-mir/src/structure.rs

```diff
@@ -0,0 +1,38 @@
+use crate::types::{visibility::Visibility, Type};
+
+/// Represents the properties of a struct.
+/// i.e. `a: B`
+#[derive(Debug, Clone)]
+pub struct StructProperty {
```

Comment:
Shouldn't this have a visibility field?

Reply:
at least in LLVM, struct fields don't have any special visibility features. Field visibility is a convenience provided by the frontend